src/libs/actions/Session/index.ts

```diff
@@ -272,6 +273,16 @@ function signOutAndRedirectToSignIn(shouldResetToHome?: boolean, shouldStashSess
                 [ONYXKEYS.SESSION]: stashedSession,
             };
         }
+
+        // Now if this is a copilot access, we do not want to stash the current session and we have a
+        // stashed session, then we need to restore the stashed session instead of completely logging out
+        if(shouldHandleCopilotSignout && shouldStashSession && hasStashedSession()){
```

Comment:
Why `shouldStashSession` must be true? I think we can just use the previous block to handle this.

```js
        if ((isSupportal || shouldForceUseStashedSession) && !shouldStashSession && hasStashedSession()) {
```